### DISCUSSION TITLE
How to simply read an EPUB file

### DISCUSSION THREAD

**Discussion by conradbekondo** [readium/architecture#180] [Q&A] [answered]
Hello there,

I'm working on a JavaFX application where I need to be able to read EPUB files.
I've come across multiple repositories, but they're all either web based or android based. My questions thus are;
- Is it possible for one to be able to read an EPUB using just Java?
- If not, then how can I adapt an existing library to meet my requirement?
- Or have I been looking in the wrong directions by studying the readium desktop repo?

Thanks for reading. I look forward for a response.

**Comment by qnga** [comment, marked as answer]
Hi!
We don't support Java and have no plan to do so. I _think_ a small wrapper around the Kotlin toolkit should make it possible to use it through Java though.
There is still an issue with Android-dependent code, but we'd love to make the Kotlin toolkit multiplatform. We could discuss further feasibility in case you might be interested in doing this work.

**Reply by conradbekondo** [reply]
Oh well. Thanks anyway
